app/src/main/java/com/sun/music61/screen/play/adapter/TracksModalAdapter.java

```diff
@@ -0,0 +1,151 @@
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.animation.AnimationUtils;
+import android.widget.ImageView;
+import android.widget.TextView;
+import com.sun.music61.R;
+import com.sun.music61.data.model.Track;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class TracksModalAdapter extends RecyclerView.Adapter<TracksModalAdapter.ViewHolder> {
+
+    private static final int POSITION_START = 0;
+
+    private List<Track> mTracks;
+    private Track mTrackPlaying;
+    private TrackModalClickListener mListener;
+
+    public void setListener(TrackModalClickListener listener) {
+        mListener = listener;
+    }
+
+    public TracksModalAdapter() {
+        mTracks = new ArrayList<>();
+    }
+
+    public void setTrackPlaying(Track trackPlaying) {
+        mTrackPlaying = trackPlaying;
+    }
+
+    public void updateData(List<Track> tracks) {
+        mTracks.clear();
+        mTracks.addAll(tracks);
+        notifyItemRangeChanged(POSITION_START, tracks.size());
+    }
+
+    public void add(Track track) {
+        mTracks.add(track);
+        notifyItemInserted(getItemCount() - 1);
+    }
+
+    public void loadMoreData(List<Track> tracks) {
+        for (Track track: tracks)
+            add(track);
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int position) {
+        View rootView = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_track_line, viewGroup, false);
+        return new ViewHolder(rootView, mListener);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int position) {
+        viewHolder.bindData(mTracks.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mTracks != null ? mTracks.size() : 0;
+    }
+
+    static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+
+        private Track mTrack;
+        private TextView mTextCount;
+        private TextView mTextSong;
+        private TextView mTextAuthor;
+        private ImageView mButtonRemove;
+        private TrackModalClickListener mListener;
+        private boolean mIsPlaying;
+
+        private ViewHolder(@NonNull View itemView, TrackModalClickListener listener) {
+            super(itemView);
+            mListener = listener;
+            mTextCount = itemView.findViewById(R.id.textCount);
+            mTextSong = itemView.findViewById(R.id.textSong);
+            mTextAuthor = itemView.findViewById(R.id.textAuthor);
+            mButtonRemove = itemView.findViewById(R.id.buttonPlay);
+            mButtonRemove.setOnClickListener(this);
+            itemView.setOnClickListener(this);
+        }
+
+        public void hoverPlaying(boolean isPlaying) {
+            mIsPlaying = isPlaying;
+            if (mIsPlaying) stylePlaying();
```

Comment:
thử xem được không e
```suggestion
           isPlaying ? stylePlaying() : styleNotPlaying();
```